metadata-ingestion/src/datahub/utilities/sqlalchemy_type_converter.py

```diff
@@ -240,3 +245,12 @@ def get_schema_fields_for_sqlalchemy_column(
     )
 
     return schema_fields
+
+
+def get_native_data_type_for_sqlalchemy_type(
+    column_type: types.TypeEngine, inspector: Inspector
+) -> str:
+    if isinstance(column_type, types.NullType):
+        return types.NullType.__visit_name__
+
+    return column_type.compile(dialect=inspector.dialect)
```

Comment:
I still think we should have a try catch around this - purely to ensure that we don't fail broadly if an underlying dialect throws an exception in `.compile`

Reply:
I have added a try/except, which will use the __visit_name__ as a fallback and in case the data type is not visitable (which the data type should be, but who knows...) the repr of the data type. I would not expect to ever need this fallback, but to make sure that we are not failing because of the native data type it's probably better to have it.